var/spack/repos/builtin/packages/fbgemm/package.py

```diff
@@ -35,4 +35,12 @@ class Fbgemm(CMakePackage):
     depends_on('python', type='build')
     depends_on('llvm-openmp', when='%apple-clang')
 
+    conflicts('%gcc@:4', msg='FBGEMM requires GCC 5+')
+
     generator = 'Ninja'
+
+    @run_before('cmake')
+    def check_requirements(self):
+        if 'avx2' not in self.spec.target:
+            raise RuntimeError(
+                'FBGEMM requires a CPU with support for AVX2 instruction set or higher')
```

Comment:
@adamjstewart I think we can get the same effect during concretization by:
```python
conflicts('target=x86_64:x86_64_v3')
```
even though technically we would need an open interval (since `x86_64_v3` would be the first uarch allowed).

Reply:
I would love to get the same effect during concretization, but that target range is completely illegible to me...

If there was a "not" in Spack's spec syntax it would help. How hard would it be to add a `requires` directive that is the opposite of `conflicts`?

Reply:
> I would love to get the same effect during concretization, but that target range is completely illegible to me...

It's just based on https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels :slightly_smiling_face: The only weak point is that technically `x86_64_v3` should be the first allowed value.

Reply:
In that case should it be:
```python
conflicts('target=x86_64:x86_64_v2')
```
?

Reply:
No, since that will allow uarch with AVX, but not AVX2 to be selected.

Reply:
![cpu](https://user-images.githubusercontent.com/4199709/122455394-a2590680-cfac-11eb-8721-b2587007b633.png)

Reply:
It sounds like this approach isn't very reliable so I'll just keep it as is until we can do something like:
```python
requires('target=avx2')
```